torch/linalg/__init__.py

```diff
@@ -88,6 +88,49 @@
 Alias of :func:`torch.det`.
 """)
 
+matrix_rank = _add_docstr(_linalg.linalg_matrix_rank, r"""
+matrix_rank(input, tol=None, hermitian=False) -> Tensor
+
+Computes the numerical rank of a matrix :attr:`input`,
+or of each matrix in a batched :attr:`input`. The method to compute the
+matrix rank is done using SVD by default. If :attr:`hermitian` is ``True``,
```

Comment:
This is a very good initial paragraph, but I think we can tweak it slightly.

The first sentence is excellent, but the discussion of how the computation is performed should come later in the documentation, I think. The second sentence can define what the rank of the matrix is using `tol`. 

Then the second paragraph can elaborate on how `tol` is computed if it's None, and finally a third paragraph can discuss how this is computed and what the `hermitian` argument does. When describing the method of computation, don't start with an acronym, but the full name of the method (i.e. "singular value decomposition) and reference the corresponding. While we're not actually using the newer linear algebra functions internally we should still reference them here.

Reply:
I've updated now the description:
> Computes the numerical rank of a matrix :attr:`input`, or of each matrix in a batched :attr:`input`.
The matrix rank is computed as the number of singular values (or the absolute eigenvalues when :attr:`hermitian` is ``True``)
above the specified :attr:`tol` threshold.

> If :attr:`tol` is not specified, :attr:`tol` is set to
``S.max(dim=-1) * max(input.shape[-2:]) * eps`` where ``S`` is the singular values
(or the absolute eigenvalues when :attr:`hermitian` is ``True``),
and ``eps`` is the epsilon value for the datatype of :attr:`input`.

> The method to compute the matrix rank is done using singular value decomposition (see :func:`torch.linalg.svd`) by default.
If :attr:`hermitian` is ``True``, then :attr:`input` is assumed to be Hermitian (symmetric if real-valued),
and the computation of the rank is done by obtaining the eigenvalues (see :func:`torch.linalg.eigvalsh`).